frontend/ui/uimanager.lua

```diff
@@ -661,16 +667,13 @@ end
 
 --- Get top widget (name if possible, ref otherwise).
 function UIManager:getTopWidget()
-    if not self._window_stack[1] then
+    if #self._window_stack < 1 then
```

Comment:
A single access/nil check is faster than walking the list to count it, even with LuaJIt's size gint mechanism ;).

Reply:
Thanks, good point I was not aware that the `#` operator has a O(n) runtime complexity. 

I will change every `#xxx` usage to a `xxx.len` property, so that the complexity reduces to O(1) ;-)

Reply:
Joking aside. Would it then make sense to change
https://github.com/koreader/koreader/blob/60a16c7e00ac6ac024ef32b702839f95c070d3a2/frontend/ui/uimanager.lua#L683-L687
to 
```
function UIManager:getSecondTopmostWidget()
    if not self._window_stack[2] then
        -- Not enough widgets in the stack, bye!
        return nil
    end
```

Reply:
That's much less readable (and this isn't a hot path anyway), so, no ;).